notification_lib/novu_manager/generic_novu_manager.py

```diff
@@ -51,3 +51,50 @@ def get_generic_novu_templates(cls) -> list[dict[str, Any]]:
         logging.error(f"status code: {response.status_code}")
         logging.error(f"response : {response.json()}")
         raise NotificationException("Erreur lors de l'importation des templates")
+
+    @staticmethod
+    def format_filter(step_filter: dict[str, Any]) -> dict[str, Any]:
+        """Take useful fields for the filter."""
+
+        needed_fields = ["children", "isNegated", "type", "value"]
+        formated_filter = {}
+        for key in step_filter.keys():
+            if key in needed_fields:
+                formated_filter[key] = step_filter[key]
+        return formated_filter
```

Comment:
return {key: step_filter[key] for key in step_filter.keys() if key in ["children", "isNegated", "type", "value"]}